HotelBooking/HotelBooking.BusinessLogic/Services/Implementation/GuestService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using FluentValidation.Results;
+using HotelBooking.BusinessLogic.Dtos;
+using HotelBooking.BusinessLogic.Exceptions;
+using HotelBooking.BusinessLogic.Services.Abstraction;
+using HotelBooking.BusinessLogic.Validators;
+using HotelBooking.DataAccessLayer.Entities;
+using HotelBooking.DataAccessLayer.Repositories.Interfaces;
+using System.Text;
+
+namespace HotelBooking.BusinessLogic.Services.Implementation;
+public class GuestService : IGuestService
+{
+    private readonly IGuestRepository _guestRepository;
+    private readonly IUnitOfWork _unitOfWork;
+
+    public GuestService(IGuestRepository guestRepository, IUnitOfWork unitOfWork)
+    {
+        _guestRepository = guestRepository;
+        _unitOfWork = unitOfWork;
+    }
+
+    public async Task AddGuest(GuestDto guestDto)
+    {
```

Comment:
ten cały kod od linijki 24 do 39 raczej będzie się powtarzał dla wielu endpointów w przyszłości. Ja bym go wydzielił do osobnej klasy. Stworzyłbym w naszych serwisach jakiś ValidationService i w nim umieścił ten kod. Następniu tutaj w GuestService wstrzyknął go sobie przez konstruktor i wywołał coś w stylu `await _validationService.Validate(guestDto)`. Żeby to fajnie działało też dla kolejnych endpointów ten nowy ValidationService powinien być generyczny:
`internal class ValidationService<TRequest> : IValidationService<TRequest> where TRequest : class`
i w nim jakaś metoda, która już będzie korzystać z tego validatora z fluentValidation:
`public async Task Validate(TRequest request)`
Ona wykona to co masz tutaj od linijki 24 do 39. Różnica będzie taka, że zamiast tworzyć ten validator przez new wstrzykniemy go sobie przez konstruktor jako `IValidator<TRequest>`